schema/samcli.json

```diff
@@ -187,9 +187,140 @@
+                  "title": "no_event",
+                  "type": "boolean",
+                  "description": "DEPRECATED: By default no event is assumed.",
+                  "default": true
+                },
+                "template_file": {
+                  "title": "template_file",
+                  "type": "string",
+                  "description": "AWS SAM template which references built artifacts for resources in the template. (if applicable)",
+                  "default": "template.[yaml|yml|json]"
+                },
+                "env_vars": {
+                  "title": "env_vars",
+                  "type": "string",
+                  "description": "JSON file containing values for Lambda function's environment variables."
+                },
+                "parameter_overrides": {
+                  "title": "parameter_overrides",
+                  "type": "string",
+                  "description": "String that contains AWS CloudFormation parameter overrides encoded as key=value pairs."
+                },
+                "debug_port": {
+                  "title": "debug_port",
+                  "type": "integer",
+                  "description": "When specified, Lambda function container will start in debug mode and will expose this port on localhost."
+                },
+                "debugger_path": {
+                  "title": "debugger_path",
+                  "type": "string",
+                  "description": "Host path to a debugger that will be mounted into the Lambda container."
+                },
+                "debug_args": {
+                  "title": "debug_args",
+                  "type": "string",
+                  "description": "Additional arguments to be passed to the debugger."
+                },
+                "container_env_vars": {
+                  "title": "container_env_vars",
+                  "type": "string",
+                  "description": "JSON file containing environment variables to be set within the container environment"
+                },
+                "docker_volume_basedir": {
+                  "title": "docker_volume_basedir",
+                  "type": "string",
+                  "description": "Specify the location basedir where the SAM template exists. If Docker is running on a remote machine, Path of the SAM template must be mounted on the Docker machine and modified to match the remote machine."
+                },
+                "log_file": {
+                  "title": "log_file",
+                  "type": "string",
+                  "description": "File to capture output logs."
+                },
+                "layer_cache_basedir": {
+                  "title": "layer_cache_basedir",
+                  "type": "string",
+                  "description": "Specify the location basedir where the lambda layers used by the template will be downloaded to.",
+                  "default": "/Users/leogama/.aws-sam/layers-pkg"
```

Comment:
It looks like the default value for it gets computed using `click.get_app_dir()` in [this section of the codebase](https://github.com/aws/aws-sam-cli/blob/develop/samcli/commands/local/cli_common/options.py#L12-L35). Since it's always calling `.joinpath()` with that last folder, **layers-pkg**, we could either parse that out in the generation as the default, or get rid of the default value altogether. Either way, this will probably need to be explicitly excepted as a special case.